### DISCUSSION TITLE
floating disable does not work for me

### DISCUSSION THREAD

**Discussion by AdamWysokinski** [i3/i3#5621] [Q&A] [answered]
Hi,
I'd like to disable floating for some windows, e.g. Firefox. I have this in my config:
```
bindsym $mod+space floating toggle
for_window [class="firefox-beta" title=".*Firefox.*"] floating disable
```
$mod+space makes the Firefox window floating. Can I prevent it?

**Comment by okraits** [comment, marked as answer]
`for_window` directives are applied only when new windows are created, not when windows are manipulated by commands.

You need to add [command criteria](https://i3wm.org/docs/userguide.html#command_criteria) to your keybinding for `floating toggle`. I guess you can use a regular expression to exclude the firefox window.

**Reply by AdamWysokinski** [reply]
Thanks, I'll take a look. But in this case this does not solve my problem, but I'll submit this as a new issue.

**Comment by okraits** [comment]
> Thanks, I'll take a look. But in this case this does not solve my problem, but I'll submit this as a new issue.

Why not?

**Reply by AdamWysokinski** [reply]
Because my problem is related to already opened windows that become randomly floating (see https://github.com/i3/i3/issues/5623). I wanted to make them un-floatingable as an intermediate solution.
